include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h

```diff
@@ -922,7 +923,7 @@ void EndParallelSectionInternal(PerThread &pt,
 
   // Wait for workers to exit ParLoopWorker
   auto tasks_to_wait_for = tasks_started - tasks_revoked;
-  while (ps.tasks_finished < tasks_to_wait_for) {
+  while (ps.tasks_finished.load(std::memory_order_acquire) < tasks_to_wait_for) {
```

Comment:
Do we have evidence that we benefit from adding additional controls over memory ordering?  It adds complexity.

Reply:
Change restored, will test with that separately.